### ISSUE TITLE
Crash on expressionProperties if 'templateOptions.options' stream source is a subject

### ISSUE THREAD

**Issue by Greegko** [ngx-formly/ngx-formly#2922]
**Description**

For `templateOptions.options`!

When the observable is `interval` or `from` works perfectly, but when we adjust to `Subject` or `BehaviourSubject` it crashes, because the `JSON.stringify` crashes on circular references.

Super short crashing snippet:
```
expressionProperties: {
  'templateOptions.options': modal => new Subject()
}
```

When we target other `templateOptions` like `templateOptions.label` it is working fine.

**Minimal Reproduction**

https://stackblitz.com/edit/ngx-formly-observable-options-crash

- Angular version: stackblitz with 9.x, we use 11.x
- Formly version: 5.10.21

Find it a nice way to recreate the observable when the target module path value is changing, therefore we would like to use with difference sources and map / filter based on the model actual values.

**Comment by kenisteward** [comment]
Have you tried returning

 new Subject().asObservable()?

**Comment by Greegko** [comment]
> Have you tried returning new Subject().asObservable()?

Sure I am not using pure Subject in general, we are always doing some operations, therefore pipe would create an observable anyway, but even with `asObservable` it won't change the fact the source is a Subject, just hide the interfaces to emit values. But for fun, tried but doesn't help.

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.22](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.22) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
